src/algebra/poly/authenticated_poly.rs

```diff
@@ -379,6 +476,65 @@ impl<C: CurveGroup> Div<&DensePolynomialResult<C>> for &AuthenticatedDensePoly<C
 }
 impl_borrow_variants!(AuthenticatedDensePoly<C>, Div, div, /, DensePolynomialResult<C>, C: CurveGroup);
 
+/// Authenticated division, i.e. division in which the divisor is a secret shared polynomial
+///
+/// We follow the approach of: https://iacr.org/archive/pkc2006/39580045/39580045.pdf (Section 5)
+///
+/// To see why this method holds, consider the `rev` operation for a polynomial a(x):
+///     rev(a) = x^deg(a) * a(1/x)
+/// Note that this operation is equivalent to reversing the coefficients of a(x)
```

Comment:
See my comment below, could we define this as in the paper:
```suggestion
///     rev_k(a) = x^k * a(1/x)
/// Note that this operation is equivalent to reversing the coefficients of a(x) if deg(a) == k
```